odoo/addons/base/models/res_lang.py

```diff
@@ -381,7 +374,7 @@ def format(self, percent, value, grouping=False, monetary=False):
 
         # floats and decimal ints need special action!
         if grouping:
-            lang_grouping, thousands_sep, decimal_point = self._data_get(monetary)
+            lang_grouping, thousands_sep, decimal_point = self.grouping, self.thousands_sep or '', self.decimal_point
```

Comment:
You could actually retrieve those values with `get_all_info_by('id')[self.id]` and take advantage of the ormcache, _provided those extra fields are included in `LangInfo`_.